docs/go-selectors.md

```diff
@@ -0,0 +1,120 @@
+
+## Full API Example
+
+The API is sufficiently different from `select` that it bears documenting:
+
+```go
+func SampleWorkflow(ctx workflow.Context) error {
+	// standard Workflow setup code omitted...
+
+	// API Example: declare a new selector
+	selector := workflow.NewSelector(ctx)
+
+	// API Example: defer code execution until the Future that represents Activity result is ready
+	work := workflow.ExecuteActivity(ctx, ExampleActivity)
+	selector.AddFuture(work, func(f workflow.Future) {
+		// deferred code omitted...
+	})
+
+	// more parallel timers and activities initiated...
+
+	// API Example: receive information from a Channel
+	var signalVal string
+	channel := workflow.GetSignalChannel(ctx, channelName)
+	selector.AddReceive(channel, func(c workflow.ReceiveChannel, more bool) {
+		// matching on the channel doesn't consume the message.
+    // So it has to be explicitly consumed here
+		c.Receive(ctx, &signalVal)
+		// do something with received information
+	})
+
+	// API Example: block until the next Future is ready to run
+  // important! none of the deferred code runs until you call selector.Select
+	selector.Select(ctx)
+
+	// Todo: document selector.HasPending
+}
+```
+
+## Using Selectors with Futures
+
+You add `Future`s after `Activities`:
+
+```go
+	// API Example: defer code execution until after an activity is done
+	work := workflow.ExecuteActivity(ctx, ExampleActivity)
+	selector.AddFuture(work, func(f workflow.Future) {
+		// deferred code omitted...
+	})
+```
+
+`selector.Select(ctx)` is the primary mechanism which blocks on and executes `Future` work.
+It is intentionally flexible; you may call it conditionally or multiple times:
+
+```go
+	// API Example: blocking conditionally
+  if somecondition != nil {
+		selector.Select(ctx)
+  }
+
+	// API Example: popping off all remaining Futures
+  for i := 0; i < len(someArray); i++ {
+		selector.Select(ctx) // this will wait for one branch
+		// you can interrupt execution here
+	}
+```
+
+A Future matches only once per Selector instance even if Select is called multiple times.
+If multiple items are available, the order of matching is not defined.
+
+## Using Selectors with Channels
+
+`selector.AddReceive(channel, func(c workflow.ReceiveChannel, more bool) {})` is the primary mechanism which receives messages from `Channels`.
+
+```go
+	// API Example: receive information from a Channel
+	var signalVal string
+	channel := workflow.GetSignalChannel(ctx, channelName)
+	selector.AddReceive(channel, func(c workflow.ReceiveChannel, more bool) {
+		c.Receive(ctx, &signalVal)
+		// do something with received information
+	})
+```
+
+Merely matching on the channel doesn't consume the message; it has to be explicitly consumed with a `c.Receive(ctx, &signalVal)` call.
+
+## Querying Selector State
+
+TODO: document `selector.HasPending`
+Context: This was added to ensure that signals are not lost when Workflow closed.
+Explain how to ensure signals are not lost.
```

Comment:
any thoughts on whether/how to document this?

Reply:
Maybe, but right now its making this pr stale -

Reply:
Made a task - lets tackle this in another iteration -
What else needs to happen to get this into the docs?

Reply:
i dont see anything else blocking IMO